### PR TITLE
changed B3Propagator to default to inject/extra single header

### LINKED ISSUE #1684: Create a combined B3 Propagator that handle both single and multi-header cases
See the recently updated spec:  https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/context/api-propagators.md#b3-requirements

### PR DESCRIPTION
**Pull request by jarebudev**
resolves #1684 

Wasn't sure what you had in mind for making the ```B3Propagator``` configurable so I added a builder that can configure creation of it.

Default (inject single header)
```
B3Propagator.getInstance();
```

Using builder
```
    B3Propagator.builder()
      .injectFromMultipleHeaders()
      .build();
```

### CODE REVIEW

File: extensions/trace_propagators/src/main/java/io/opentelemetry/extensions/trace/propagation/B3PropagatorExtractor.java [hunk 1]
```diff
@@ -7,10 +7,12 @@
 
 import io.grpc.Context;
 import io.opentelemetry.context.propagation.TextMapPropagator;
+import javax.annotation.Nullable;
 import javax.annotation.concurrent.Immutable;
 
 @Immutable
 interface B3PropagatorExtractor {
 
+  @Nullable
```
Comment (jkwatson) [inline extensions/trace_propagators/src/main/java/io/opentelemetry/extensions/trace/propagation/B3PropagatorExtractor.java:16]: Any thoughts about using `Optional<Context>` as the return type, now that we have java 8?

Comment (jarebudev) [reply extensions/trace_propagators/src/main/java/io/opentelemetry/extensions/trace/propagation/B3PropagatorExtractor.java:16]: Nice! I'm glad Java 8 is supported now :)
Yeh returning an ```Optional``` would be better than null.  I'll update it.

**Review by jkwatson — APPROVED** [review]
Thanks!

**Review by thisthat — APPROVED** [review]
Thank you :)

File: extensions/trace_propagators/src/main/java/io/opentelemetry/extensions/trace/propagation/B3Propagator.java [hunk 2]
```diff
@@ -31,44 +58,54 @@
...
-  public static B3Propagator getMultipleHeaderPropagator() {
-    return MULTI_HEADER;
+  public static class Builder {
+    private boolean injectSingleHeader;
+
+    private Builder() {
+      injectSingleHeader = true;
+    }
+
+    public Builder injectFromMultipleHeaders() {
```
Comment (anuraaga) [inline extensions/trace_propagators/src/main/java/io/opentelemetry/extensions/trace/propagation/B3Propagator.java:92]: Inject happens to headers, not from. But how about just `injectMultipleHeaders`?

**Review by anuraaga — APPROVED** [review]
Just nit thanks!

File: extensions/trace_propagators/src/main/java/io/opentelemetry/extensions/trace/propagation/B3Propagator.java (hunk 2 above)
Comment (jarebudev) [reply extensions/trace_propagators/src/main/java/io/opentelemetry/extensions/trace/propagation/B3Propagator.java:92]: sounds good, i'll tidy it up

**Pull request merged**